packages/core/lib/config.js

```diff
@@ -10,40 +10,44 @@ const isPlainObject = require('is-plain-object');
 const escapeRegExp = require('escape-string-regexp');
 const flatten = require('flat');
 
-const merge = (...args) =>
-  mergeWith({}, ...args, (objValue, srcValue, key) => {
-    if (Array.isArray(objValue)) {
-      if ('mixins' === key) {
-        return objValue.concat(srcValue);
-      }
-      return srcValue;
-    }
-  });
-
-const resolvePreset = (context, preset) => {
-  return createResolver({
-    mainFiles: ['preset'],
-    mainFields: ['preset'],
-  })(context, preset);
-};
+const resolvePreset = createResolver({
+  mainFiles: ['preset'],
+  mainFields: ['preset'],
+});
+
+const resolveCoreMixin = createResolver({
+  mainFiles: ['mixin.core', 'mixin'],
+  mainFields: ['mixin:core', 'mixin'],
+});
+
+const resolveServerMixin = createResolver({
+  mainFiles: ['mixin.server', 'mixin.runtime', 'mixin'],
+  mainFields: ['mixin:server', 'mixin:runtime', 'mixin'],
+});
+
+const resolveBrowserMixin = createResolver({
+  mainFiles: ['mixin.browser', 'mixin.runtime', 'mixin'],
+  mainFields: ['mixin:browser', 'mixin:runtime', 'mixin'],
+});
 
 const resolveMixin = (target, context, mixin) => {
   try {
-    const config = {
-      mainFiles: [`mixin.${target}`, 'mixin'],
-      mainFields: [`mixin:${target}`, 'mixin'],
-    };
-    if (target !== 'core') {
-      config.mainFiles.splice(1, 0, 'mixin.runtime');
-      config.mainFields.splice(1, 0, 'mixin:runtime');
+    switch (target) {
+      case 'core':
+        return resolveCoreMixin(context, mixin);
+      case 'server':
+        return resolveServerMixin(context, mixin);
+      case 'browser':
+        return resolveBrowserMixin(context, mixin);
+      default:
+        throw new Error();
```

Comment:
I'd prefer to either `return null` here in the `default` or throw a more meaningful error.